rbac/management/group/definer.py

```diff
@@ -166,12 +179,11 @@ def add_roles(group, roles_or_role_ids, tenant, user=None):
             continue
 
         system_policy.roles.add(role)
-
-        if tenant.tenant_name != "public":
-            dual_write_handler = RelationApiDualWriteGroupHandler(group, ReplicationEventType.ASSIGN_ROLE)
-            dual_write_handler.replicate_added_role(role)
-        # Send notifications
         group_role_change_notification_handler(user, group, role, "added")
+        if dual_write_handler:
+            dual_write_handler.replicate_added_roles([role])
+    if dual_write_handler:
+        dual_write_handler.replicate()
```

Comment:
GitHub isn't letting me make the full suggestion, but with the method that supports a list, maybe we want to do this:

```
    new_roles: list[Role] = []

    for role in [*system_roles, *custom_roles]:
        # Only Organization administrators are allowed to add the role with RBAC permission
        # higher than "read" into a group.
        for access in role.access.all():
            if (
                access.permission_application() == "rbac"
                and access.permission.verb != "read"
                and user
                and not user.admin
            ):
                key = "add-roles"
                message = (
                    "Non org admin users are not allowed to add RBAC role with higher than 'read' permission "
                    "into groups."
                )
                raise serializers.ValidationError({key: _(message)})

        # Only add the role if it was not attached
        if system_policy.roles.filter(pk=role.pk).exists():
            continue

        system_policy.roles.add(role)
        group_role_change_notification_handler(user, group, role, "added")
        new_roles.append(role)
    
    if group.tenant.tenant_name != "public":
        dual_write_handler = RelationApiDualWriteGroupHandler(group, ReplicationEventType.ASSIGN_ROLE)
        dual_write_handler.replicate_added_roles(new_roles)
        dual_write_handler.replicate()
```

See other comment wrt `replicate_added_roles` name

I think this should really be either:

```
        dual_write_handler.add_roles(new_roles)
        dual_write_handler.replicate()
```

or just

```
        dual_write_handler.replicate_added_roles(new_roles)
```